src/algebra/star/basic.lean

```diff
@@ -291,6 +291,12 @@ lemma star_ring_end_apply [comm_semiring R] [star_ring R] {x : R} :
 @[simp] lemma star_ring_end_self_apply [comm_semiring R] [star_ring R] (x : R) :
   star_ring_end R (star_ring_end R x) = x := star_star x
 
+instance {S : Type u} [non_assoc_semiring S] [comm_semiring R] [star_ring R] :
+  has_involutive_star (S →+* R) :=
```

Comment:
Can you add `star_def` and `star_apply` lemmas?

Reply:
I have added the `star_def` lemma. When I had the `star_apply` lemma:
```lean
@[simp]
lemma ring_hom.star_apply {S : Type u} [non_assoc_semiring S] [comm_semiring R] [star_ring R]
  (f : S →+* R) (s : S) : star f s = star (f s) := rfl
```
I get an error from `lint`:
```
/- SOME SIMP LEMMAS ARE NOT IN SIMP-NORMAL FORM.
#check @ring_hom.star_apply /- Left-hand side simplifies from
  ⇑(star f) s
to
  ⇑conj (⇑f s)
using 
  [ring_hom.star_def, function.comp_app, ring_hom.coe_comp]
Try to change the left-hand side to the simplified term!
 -/
so maybe this lemma is not needed after all? (I left it for the time being)